### PR TITLE
Export, import and logout

### PR DESCRIPTION
**Pull request by oktaysenkan**
## Features

- Added exporting all password .csv format. #1 
- Added import passwords .csv format. #3
- Logout is added.

## Screenshots

image

### CODE REVIEW

File: components/actions-menu.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const Actions = ({
+  onImport = () => {},
+  onExport = () => {},
+  onLogout = () => {},
+}) => {
+  const fileInput = React.useRef();
+
+  const handleImport = () => {
+    fileInput.current.click();
+  };
```
Comment (yasintz) [inline components/actions-menu.js:18]: Using `React.useCallback` has good for performance.

File: components/actions-menu.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <Menu className="dropdown-menu">
+      <Menu.Item onClick={handleImport}>
+        <ImportOutlined />
+        <input
+          type="file"
+          id="file"
+          ref={fileInput}
+          style={{ display: "none" }}
+          accept=".csv"
+          onChange={(e) => onImport(e.target.files[0])}
```
Comment (yasintz) [inline components/actions-menu.js:30]: Using React.useCallback has good for performance.
example
```js
const handleFileChange = React.useCallback((e) => onImport(e.target.files[0]), [onImport])
```
...
```js
onChange={handleFileChange}
```

File: pages/index.js [hunk 3]
```diff
@@ -24,6 +26,55 @@ function HomePage() {
     message.error(error)
   }, [error])
 
+  const handleLogout = () => {
```
Comment (yasintz) [inline pages/index.js:29]: Using `React.useCallback` has good for performance.

File: pages/index.js [hunk 4]
```diff
@@ -24,6 +26,55 @@ function HomePage() {
...
+    localStorage.removeItem('TOKEN')
+    Router.replace('/login')
+  }
+
+  const handleExport = async () => {
```
Comment (yasintz) [inline pages/index.js:34]: Using `React.useCallback` has good for performance.

File: pages/index.js [hunk 5]
```diff
@@ -24,6 +26,55 @@ function HomePage() {
...
+
+      message.success('Passwords exported')
+      revalidate()
+    } catch (e) {
+      console.log(e)
+      message.error(e.message)
+    }
+  }
+
+  const handleImport = async (file) => {
```
Comment (yasintz) [inline pages/index.js:54]: Using `React.useCallback` has good for performance.

File: components/actions-menu.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+const Actions = ({
+  onImport = () => {},
+  onExport = () => {},
+  onLogout = () => {},
+}) => {
+  const fileInput = React.useRef();
+
+  const handleImport = () => {
+    fileInput.current.click();
```
Comment (yasintz) [inline components/actions-menu.js:17]: `fileInput.current` may be undefined.
```js
if(fileInput.current){
...
}
```

File: pages/index.js [hunk 7]
```diff
@@ -24,6 +26,55 @@ function HomePage() {
...
+        {
+          method: 'POST'
+        },
+        false
+      )
+
+      download(data, 'passwall.csv', 'text/csv')
+
+      message.success('Passwords exported')
+      revalidate()
```
Comment (yasintz) [inline pages/index.js:47]: The export process does not cause a change on `logins`. It doesn't have to be called.

**Comment by yakuter** [comment]
Hello @oktaysenkan sorry for late response. Good shares. Please tell me when you reslove conflicts then I will merge this PR. Thank you so much

**Comment by yakuter** [comment]
Great job. Thank you.

**Pull request merged**